### PR TITLE
sratoolkit: init at 2.11.3

### PR DESCRIPTION
**Pull request by thyol**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/sratoolkit/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{ stdenv, lib
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/sratoolkit/default.nix:1]:
```suggestion
{ stdenv
, lib
```
This was just an error in a treewide change

File: pkgs/applications/science/biology/sratoolkit/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+, fetchurl
+, autoPatchelfHook
+, libidn
+, zlib
+, bzip2
+}:
+
+stdenv.mkDerivation rec {
+  name = "sratoolkit";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/sratoolkit/default.nix:11]:
```suggestion
```

File: pkgs/applications/science/biology/sratoolkit/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
+{ stdenv, lib
+, fetchurl
+, autoPatchelfHook
+, libidn
+, zlib
+, bzip2
+}:
+
+stdenv.mkDerivation rec {
+  name = "sratoolkit";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/sratoolkit/default.nix:10]:
```suggestion
  pname = "sratoolkit";
```

File: pkgs/applications/science/biology/sratoolkit/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  version = "2.11.3";
+
+  libidn11 = libidn.overrideAttrs (old: {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/sratoolkit/default.nix:14]: Do we really need this? Can we patch instead? If not then please move this at least in a let in. Also please use pname+version.

File: pkgs/applications/science/biology/sratoolkit/default.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    libidn11
+    zlib
+    bzip2
+    stdenv.cc.cc.lib
+  ];
+
+  sourceRoot = "./sratoolkit.${version}-ubuntu64/bin";
+
+  installPhase = ''
+    for i in $(ls -p | grep -v / | grep -v remote-fuser ) ;  do
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/sratoolkit/default.nix:41]:
```suggestion
    for i in $(ls -p | grep -v / | grep -v remote-fuser);  do
```
Please do not rely on ls. How many files do we need to copy? Maybe just do it manually.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 4 above)
Comment (thyol) [reply pkgs/applications/science/biology/sratoolkit/default.nix:12]: Could you please elaborate what you mean with can we patch instead? We cannot use a newer version without patching somehow a binary.

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 5 above)
Comment (thyol) [reply pkgs/applications/science/biology/sratoolkit/default.nix]: Please review the new version, I don't know if this is really better. The new version has ~100 lines more.

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/biology/sratoolkit/default.nix:12]: I thought we would be building from source

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/biology/sratoolkit/default.nix]: Thats not really better. Can we use find to search for executables?

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 4 above)
Comment (thyol) [reply pkgs/applications/science/biology/sratoolkit/default.nix:12]: The effort to build this from the sources is disproportionate.

Comment (SuperSandro2000) [reply pkgs/applications/science/biology/sratoolkit/default.nix:12]: What makes it so hard?

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 5 above)
Comment (thyol) [reply pkgs/applications/science/biology/sratoolkit/default.nix]: Yes, looks better.

File: pkgs/applications/science/biology/sratoolkit/default.nix (hunk 4 above)
Comment (thyol) [reply pkgs/applications/science/biology/sratoolkit/default.nix:12]: For building, we would need ncbi-vdb and ngs. ncbi-vdb and ngs depend on parts on each other. Both require a couple of libraries (~20, from my mind, could be wrong). Many are not in nixpkgs. I have put more than 8h to building this, and failed on more than basic libraries.

**Pull request merged**